file-formats/intm/examples/sapj97myt5r25l0v2ijp2rt1pko4.intm.json

```diff
@@ -0,0 +1,35 @@
+{
+  "formatVersion": "1",
+  "header": {
+    "description": "Intelligent Scenario Model via ADT",
+    "originalLanguage": "en"
+  },
+  "generalInformation": {
+    "modelName": "RM_TEST_ADT_976_MOD",
+    "modelType": "SAPGENAI",
+    "intelligentScenario": "RM_TEST_ADT_976",
+    "modellingContextId": "7FD071C0C04C56148FE3424E30FFB40B",
+    "adapterId": "SAPGENAI.Default",
+    "nativeModelType": "SAPGENAI.Default",
+    "state": "Published",
+    "executableId": "azure-openai",
+    "llmName": "gpt-4",
+    "llmVersion": "0613"
+  },
+  "modelPromptTemplates": [
+    {
+      "promptTemplateName": "SYSTEM_PROMPT",
+      "promptTemplateDescription": "System prompt",
+      "prompt": "You are a sales executive who is responsible for reaching out to customers for confirming or declining their orders. Evaluate the size of the delivery vehicle based on the number of items. Calculate the distance between the billing and delivery location in kilometers. \r\n\r\nThe subject of email should be of format <Confirmation on order number: {order_number} placed on {booking_date}> \r\nThe body of email should be of format: \"\"SO contains <Category> to be delivered to <Location>. And will be delivered via a <size> vehicle. \r\nIF <distance between source address and delivery address> > 100 kms THEN include a message about potential delivery delays ELSE congratulate on being eligible for one day delivery",
+      "promptParameters": "[]",
+      "isVisible": true
+    },
+    {
+      "promptTemplateName": "USER_PROMPT",
+      "promptTemplateDescription": "User prompt",
+      "prompt": "Build a confirmation mail for a sales order containing {ISLM_Items} where the items are categorized for example pencil and paper will be stationary, printer and mouse will be electronics. The source address contains {ISLM_Source} and the delivery address is in {ISLM_Destination}.",
+      "promptParameters": "[{\"Name\":\"ISLM_Items\",\"Value\":\"\"},{\"Name\":\"ISLM_Source\",\"Value\":\"\"},{\"Name\":\"ISLM_Destination\",\"Value\":\"\"}]",
```

Comment:
Related to https://github.com/SAP/abap-file-formats/pull/671#discussion_r1829153770. If it is always name-value or an empty list , it could look like this in AFF:

```suggestion
      "promptParameters": [
        { "Name": "ISLM_Items", 
           "Value": "" 
        },
        { "Name": "ISLM_Source", 
           "Value ": ""
        },
        { "Name ": "ISLM_Destination",
           "Value": ""
        }
      ]",
```

Reply:
Can be discussed next week

Reply:
Actually, our logic in AFF persistence class only builds the JSON, which is necessary for passing the parameters in the prompt execution.
<img width="571" alt="image" src="https://github.com/user-attachments/assets/33521cc4-6a8a-4d46-842f-75d1edf27a74">

<img width="812" alt="image" src="https://github.com/user-attachments/assets/6b81a94c-6f38-4209-831e-c8c2d01bdb66">

Does AFF/ADT framework stringify the JSON in the source?

Also another comment on this VALUE field - This is empty as of now, but as I mentioned earlier, we will definitely implement the default values option as user input in the ADT UI. Similar to our Fiori UI,
<img width="762" alt="image" src="https://github.com/user-attachments/assets/4a24c6ef-a64b-4858-89b7-ac7025e16edb">
So, this default parameters is future scope for ADT.

Reply:
> Does AFF/ADT framework stringify the JSON in the source? I mean the / in the output.

Since you put JSON in a string field consisting of certain characters (to be precise: quotes (`"`)), these quotes must be escaped by `\`. 

Nevertheless, even without escaping, it would be more nice if you don't put the JSON into a string, but just using it as real JSON. Still, I am not sure whether this is possible in your case, but maybe it's worth to discuss. 

Did you get my idea?

Reply:
Yes, I got your point that you want to keep it as real JSON. We kept it as string because we do store the prompt parameters in DB as a string.
If we keep it as real JSON, then we need to serialize it and store in the DB.
Also, AFF has to be changed for prompt_parameters field, which needs to be declared as a deep table within prompts table. Is it possible?

Reply:
Sorry, I didn't manage to answer earlier. :(

Yes, I think you would need to serialize it, if you would change it. I am still not 100% sure whether it is feasible at all to follow my suggested approach in your case. But if it is feasible, I would recommend it.

Maybe, it is worth to discuss my idea in a call. Then we can see the advantages and disadvantages of both approaches. What do you think?